code/modules/surgery/detoxification.dm

```diff
@@ -0,0 +1,54 @@
+/datum/surgery/detox
+	name = "Detoxification"
+	steps = list(/datum/surgery_step/incise,
+				/datum/surgery_step/retract_skin,
+				/datum/surgery_step/incise,
+				/datum/surgery_step/clamp_bleeders,
+				/datum/surgery_step/detox,
+				/datum/surgery_step/close)
+
+	target_mobtypes = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	possible_locs = list(BODY_ZONE_CHEST)
+	requires_bodypart_type = TRUE
+	replaced_by = /datum/surgery
+	ignore_clothes = FALSE
+
+/datum/surgery/detox/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/stomach/S = target.getorganslot(ORGAN_SLOT_STOMACH)
+	if(target.stat != DEAD)	//shamelessly lifted off the revival surgery but we're looking for the same critera here, a dead, non-husked, revivable patient.
+		return FALSE
+	if(target.suiciding || target.hellbound || HAS_TRAIT(target, TRAIT_HUSK))
```

Comment:
What's wrong with detoxing these?

Reply:
I suppose the first two I don't have much of a reason not to, but husks traditionally are too damaged to draw chemicals from anyway, so I'm leaving that.